pkg/upgrade/upgrade.go

```diff
@@ -351,6 +351,46 @@ func UpdateFromLegacyVersion(cli client.Client, platform deploy.Platform, appNS
 	return nil
 }
 
+// CleanupDeprecatedFeatures deletes deprecated features by removing the corresponding FeatureTrackers
+func CleanupDeprecatedFeatures(ctx context.Context, cli client.Client, platform deploy.Platform) error {
+	if platform != deploy.ManagedRhods {
+		return nil
+	}
+
+	crd := &apiextv1.CustomResourceDefinition{}
+	if err := cli.Get(ctx, client.ObjectKey{Name: "featuretrackers.features.opendatahub.io"}, crd); err != nil {
+		return err
+	}
+
+	type ResourceToDelete struct {
+		InstanceName     string
+		GroupVersionKind schema.GroupVersionKind
+	}
+
+	toDelete := []string{
+		"opendatahub-kserve-temporary-fixes", // Workaround for RHOAIENG-165
+	}
+
+	var multiErr *multierror.Error
+	for _, name := range toDelete {
+		obj := &unstructured.Unstructured{}
+		obj.SetGroupVersionKind(schema.GroupVersionKind{Kind: "FeatureTracker", Group: "features.opendatahub.io", Version: "v1"})
```

Comment:
It would be nicer if you use our own types: https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/apis/features/v1/groupversion_info.go#L30

Something like `v1.GroupVersion.WithKind("FeatureTracker")` (assuming the right `v1` is imported).